src/types/response.rs

```diff
@@ -57,9 +23,23 @@ pub struct TxLogResponse {
     pub tx__event___tx_events: Option<Vec<Vec<String>>>,
 }
 
+#[cfg(feature = "async")]
+impl futures::future::Future for TxLogResponse {
+    type Output = TxLogResponse;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut task::Context) -> Poll<Self::Output> {
+        if self.tx___tx_id > 0 {
```

Comment:
Why have you decided to poll this way?

I mean, I do know that transaction ids on Crux are sequential, but do they return something else on error cases?

Just wondering really, this may be the best way still.

Reply:
This is the path to stabilize `async`. My plan was to add an extra field for the feature that was boolean

Reply:
How would this boolean work?

Reply:
when created its value is true, else it is false

Reply:
Wait, but we would add that value to the document? I didn't understand it well

Reply:
it would be a non serialized field

Reply:
but TLDR, we wont need futures anymore hahahahha